### PR TITLE
use dot syntax

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: src/trust_region.jl [hunk 1]
```diff
@@ -143,18 +143,18 @@ function trust_region_{T}(df::AbstractDifferentiableMultivariateFunction,
...
 
         # Ratio of actual to predicted reduction (equation 11.47 in N&W)
         A_mul_B!(r_predict, J, p)
-        Base.BLAS.axpy!(one(T), r, r_predict)
+        r_predict .+= r
         rho = (sum(abs2,r) - sum(abs2,r_new)) / (sum(abs2,r) - sum(abs2,r_predict))
 
         if rho > eta
             # Successful iteration
-            copy!(r, r_new)
```
Comment (sglyon) [inline src/trust_region.jl]: Is this any better as a broadcast operation instead of just a copy?

Comment (KristofferC) [inline src/trust_region.jl]: Yeah, this could also be broadcasted.

Comment (KristofferC) [inline src/trust_region.jl]: It is the same performance. So depends what you mean with "better".

Comment (sglyon) [inline src/trust_region.jl]: I was thinking performance .

If they are the same then I'd id say the broadcast one is clearer.

Let's merge

Comment (KristofferC) [inline src/trust_region.jl]: the broadcast version of `scale!` is slower though

**Pull request merged**